In [19]:
GraduationData = read.csv(file = 'GraduationData_for_Anova.csv')
GraduationData

Student_Percentage Rural_or_not Year Fall_semester
1  56                 1            2014 4th          
2  75                 1            2014 5th          
3  79                 1            2014 6th          
4  53                 1            2015 4th          
5  72                 1            2015 5th          
6  56                 1            2016 4th          
7  53                 0            2014 4th          
8  74                 0            2014 5th          
9  78                 0            2014 6th          
10 55                 0            2015 4th          
11 74                 0            2015 5th          
12 60                 0            2016 4th

In [20]:
GraduationData$Rural_or_not = as.factor(GraduationData$Rural_or_not)

In [21]:
GraduationData$Year = as.factor(GraduationData$Year)
GraduationData$Fall_semester = as.factor(GraduationData$Fall_semester)

In [22]:
head(GraduationData)

Student_Percentage Rural_or_not Year Fall_semester
1 56                 1            2014 4th          
2 75                 1            2014 5th          
3 79                 1            2014 6th          
4 53                 1            2015 4th          
5 72                 1            2015 5th          
6 56                 1            2016 4th

In [5]:
lmod_graduation1 = lm(Student_Percentage ~ Rural_or_not + Year + Rural_or_not:Year + Rural_or_not:Fall_semester + Fall_semester:Year, data = GraduationData)
summary(lmod_graduation1)


Call:
lm(formula = Student_Percentage ~ Rural_or_not + Year + Rural_or_not:Year + 
    Rural_or_not:Fall_semester + Fall_semester:Year, data = GraduationData)

Residuals:
         1          2          3          4          5          6          7 
 2.500e-01 -2.500e-01  0.000e+00 -2.500e-01  2.500e-01 -3.469e-17 -2.500e-01 
         8          9         10         11         12 
 2.500e-01  6.245e-17  2.500e-01 -2.500e-01 -7.633e-17 

Coefficients: (3 not defined because of singularities)
                               Estimate Std. Error t value Pr(>|t|)   
(Intercept)                     53.2500     0.6614  80.506  0.00791 **
Rural_or_not1                    2.5000     0.8660   2.887  0.21230   
Year2015                         1.5000     0.8660   1.732  0.33333   
Year2016                         6.7500     0.9682   6.971  0.09070 . 
Rural_or_not1:Year2015          -4.0000     1.0000  -4.000  0.15596   
Rural_or_not1:Year2016          -6.5000     1.3229  -4.914  0.12782   
Rural_o

In [7]:
lmod_graduation2 = lm(Student_Percentage ~ Rural_or_not + Fall_semester + Year, data = GraduationData)
summary(lmod_graduation2)


Call:
lm(formula = Student_Percentage ~ Rural_or_not + Fall_semester + 
    Year, data = GraduationData)

Residuals:
    Min      1Q  Median      3Q     Max 
-2.0000 -0.8125  0.0000  1.0000  1.7500 

Coefficients:
                 Estimate Std. Error t value Pr(>|t|)    
(Intercept)       55.0000     1.1479  47.913 5.54e-09 ***
Rural_or_not1     -0.5000     0.9789  -0.511   0.6278    
Fall_semester5th  19.5000     1.1990  16.264 3.44e-06 ***
Fall_semester6th  23.7500     1.5861  14.974 5.59e-06 ***
Year2015          -1.0000     1.1990  -0.834   0.4362    
Year2016           3.2500     1.5861   2.049   0.0864 .  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1.696 on 6 degrees of freedom
Multiple R-squared:  0.9862,	Adjusted R-squared:  0.9747 
F-statistic: 85.68 on 5 and 6 DF,  p-value: 1.702e-05


In [8]:
lmod_graduation3 = lm(Student_Percentage ~ Rural_or_not + Fall_semester, data = GraduationData)
summary(lmod_graduation3)


Call:
lm(formula = Student_Percentage ~ Rural_or_not + Fall_semester, 
    data = GraduationData)

Residuals:
    Min      1Q  Median      3Q     Max 
-2.7500 -0.9375  0.0000  0.7500  4.2500 

Coefficients:
                 Estimate Std. Error t value Pr(>|t|)    
(Intercept)        55.750      1.090  51.160 2.36e-11 ***
Rural_or_not1      -0.500      1.258  -0.397    0.701    
Fall_semester5th   18.250      1.407  12.972 1.18e-06 ***
Fall_semester6th   23.000      1.780  12.925 1.22e-06 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 2.179 on 8 degrees of freedom
Multiple R-squared:  0.9696,	Adjusted R-squared:  0.9582 
F-statistic: 84.98 on 3 and 8 DF,  p-value: 2.083e-06


In [9]:
lmod_graduation4 = lm(Student_Percentage ~ Rural_or_not, data = GraduationData)
summary(lmod_graduation4)


Call:
lm(formula = Student_Percentage ~ Rural_or_not, data = GraduationData)

Residuals:
     Min       1Q   Median       3Q      Max 
-12.6667  -9.5417   0.5833   8.7083  13.8333 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)    
(Intercept)     65.667      4.561  14.397 5.18e-08 ***
Rural_or_not1   -0.500      6.450  -0.078     0.94    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 11.17 on 10 degrees of freedom
Multiple R-squared:  0.0006005,	Adjusted R-squared:  -0.09934 
F-statistic: 0.006009 on 1 and 10 DF,  p-value: 0.9397


In [10]:
lmod_graduation5 = lm(Student_Percentage ~ Rural_or_not + Fall_semester + Rural_or_not:Fall_semester, data = GraduationData)
summary(lmod_graduation5)


Call:
lm(formula = Student_Percentage ~ Rural_or_not + Fall_semester + 
    Rural_or_not:Fall_semester, data = GraduationData)

Residuals:
   Min     1Q Median     3Q    Max 
-3.000 -1.125  0.000  1.000  4.000 

Coefficients:
                               Estimate Std. Error t value Pr(>|t|)    
(Intercept)                      56.000      1.424  39.326 1.81e-08 ***
Rural_or_not1                    -1.000      2.014  -0.497 0.637162    
Fall_semester5th                 18.000      2.252   7.995 0.000204 ***
Fall_semester6th                 22.000      2.848   7.725 0.000247 ***
Rural_or_not1:Fall_semester5th    0.500      3.184   0.157 0.880374    
Rural_or_not1:Fall_semester6th    2.000      4.028   0.497 0.637162    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 2.466 on 6 degrees of freedom
Multiple R-squared:  0.9708,	Adjusted R-squared:  0.9464 
F-statistic: 39.86 on 5 and 6 DF,  p-value: 0.0001585


In [13]:
anova(lmod_graduation3, lmod_graduation2)

Res.Df RSS   Df Sum of Sq F        Pr(>F)    
1 8      38.00 NA    NA           NA         NA
2 6      17.25  2 20.75     3.608696 0.09354412

In [14]:
anova(lmod_graduation2, lmod_graduation3) # Hence only Rural_or_not and Fall_semester is sufficient, Year not needed.

Res.Df RSS   Df Sum of Sq F        Pr(>F)    
1 6      17.25 NA     NA          NA         NA
2 8      38.00 -2 -20.75    3.608696 0.09354412

In [15]:
anova(lmod_graduation2, lmod_graduation4)

Res.Df RSS      Df Sum of Sq F        Pr(>F)      
1  6       17.250 NA        NA       NA           NA
2 10     1248.167 -4 -1230.917 107.0362 1.044924e-05

In [16]:
anova(lmod_graduation3, lmod_graduation4) # Hence only Rural_or_not is not sufficient, Fall_semester is required.

Res.Df RSS      Df Sum of Sq F       Pr(>F)      
1  8       38.000 NA        NA      NA           NA
2 10     1248.167 -2 -1210.167 127.386 8.591007e-07

In [17]:
anova(lmod_graduation5, lmod_graduation3)  # Interation term Fall_semester:Rural_or_not is not required.

Res.Df RSS  Df Sum of Sq F         Pr(>F)  
1 6      36.5 NA   NA             NA       NA
2 8      38.0 -2 -1.5      0.1232877 0.886192

In [18]:
final_graduation_model = lmod_graduation3
summary(final_graduation_model)


Call:
lm(formula = Student_Percentage ~ Rural_or_not + Fall_semester, 
    data = GraduationData)

Residuals:
    Min      1Q  Median      3Q     Max 
-2.7500 -0.9375  0.0000  0.7500  4.2500 

Coefficients:
                 Estimate Std. Error t value Pr(>|t|)    
(Intercept)        55.750      1.090  51.160 2.36e-11 ***
Rural_or_not1      -0.500      1.258  -0.397    0.701    
Fall_semester5th   18.250      1.407  12.972 1.18e-06 ***
Fall_semester6th   23.000      1.780  12.925 1.22e-06 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 2.179 on 8 degrees of freedom
Multiple R-squared:  0.9696,	Adjusted R-squared:  0.9582 
F-statistic: 84.98 on 3 and 8 DF,  p-value: 2.083e-06


$\textbf{Interpretation}:$
1. The reduced model without the interations is sufficient.
2. The reduced model with only Rural_or_not and Fall_semester but not the Year is sufficient
3. The reduced model with only Rural_or_not but not the Fall_semester is NOT sufficient.
4. So our chosen model would be model3 with Rural_or_not and Fall_semester and predictors.

$\\\\$

5. The average graduation rate of NON-rural students in 4th Fall semester = 55.8% which is statistically significant.
6. Holding the semester constant, the graduation rates of rural and non-rural students, on average, are same.
7. Since the model without the Rural_or_not:Fall_semester interaction term is sufficient compared to the one with it, we conclude that the difference in graduation rates between rural and non-rural remains constant across the semesters, on average.
8. Keeping the type of student constant, the difference in graduation rate across semesters is statistically significant (makes sense).



## Next analysis would be look at the level of gender, BIPOC status, etc, as per Karen's email.